puffin/src/lib.rs

```diff
@@ -679,33 +962,44 @@ pub fn clean_function_name(name: &'static str) -> String {
 #[test]
 fn test_clean_function_name() {
     assert_eq!(clean_function_name(""), "");
-    assert_eq!(clean_function_name("foo"), "foo");
-    assert_eq!(clean_function_name("foo::bar"), "foo::bar");
-    assert_eq!(clean_function_name("foo::bar::baz"), "bar::baz");
     assert_eq!(
-        clean_function_name("some::GenericThing<_, _>::function_name"),
+        clean_function_name(&format!("foo{}", USELESS_SCOPE_NAME_POSTFIX)),
+        "foo"
+    );
+    assert_eq!(
+        clean_function_name(&format!("foo::bar{}", USELESS_SCOPE_NAME_POSTFIX)),
+        "foo::bar"
+    );
+    assert_eq!(
+        clean_function_name(&format!("foo::bar::baz{}", USELESS_SCOPE_NAME_POSTFIX)),
+        "bar::baz"
+    );
+    assert_eq!(
+        clean_function_name(&format!(
+            "some::GenericThing<_, _>::function_name{}",
+            USELESS_SCOPE_NAME_POSTFIX
+        )),
         "GenericThing<_, _>::function_name"
     );
     assert_eq!(
-        clean_function_name("<some::ConcreteType as some::bloody::Trait>::function_name"),
+        clean_function_name(&format!(
+            "<some::ConcreteType as some::bloody::Trait>::function_name{}",
+            USELESS_SCOPE_NAME_POSTFIX
+        )),
         "<ConcreteType as Trait>::function_name"
     );
 }
 
-/// Returns a shortened path to the current file.
-#[macro_export]
-macro_rules! current_file_name {
-    () => {
-        $crate::short_file_name(file!())
-    };
-}
-
 /// Shortens a long `file!()` path to the essentials.
 ///
 /// We want to keep it short for two reasons: readability, and bandwidth
 #[doc(hidden)]
 #[inline(never)]
-pub fn short_file_name(path: &'static str) -> String {
+pub fn short_file_name<'a>(path: &'a str) -> String {
```

Comment:
I big reason we shortened the file name and function name is to reduce bandwidth. Now that we only log it once, we should consider sending the full file path and the full function name, leaving it to the UI to shorten it.

We should still strip the `::{{closure}}::{{closure}}::f` suffix upfront though,
and I suggest we also remove the path up to the crate name (the folder before the `src` folder), but don't do any further path ellision

Reply:
Yes this is now done by the UI, or i store it in `ScopeDetailsOwned` which is created once for each scope. It contains the parsed names and can be read by the UI. I thought we could keep this logic it in the `report` function but at the cost that maybe first scope would be a couple ns longer.

Reply:
I can probably return a subset of the string as a static ref string, I try to keep it fully static str so that allocations are prevented.